rslib/src/scheduler/fsrs/memory_state.rs

```diff
@@ -249,18 +250,18 @@ fn get_last_revlog_info(revlogs: &[RevlogEntry]) -> HashMap<CardId, LastRevlogIn
         .into_iter()
         .for_each(|(card_id, group)| {
             let mut last_reviewed_at = None;
-            let mut last_revlog_is_manual = false;
+            let mut last_revlog_is_rescheduled = false;
             for e in group.into_iter() {
                 if e.button_chosen >= 1 {
                     last_reviewed_at = Some(e.id.as_secs());
                 }
-                last_revlog_is_manual = e.review_kind == RevlogReviewKind::Manual;
+                last_revlog_is_rescheduled = e.review_kind == RevlogReviewKind::Rescheduled;
```

Comment:
What about older cards that were rated in previous Anki versions. Should we be accepting either manual or rescheduled here?

Reply:
My concern is the user may feel confused if they set due date  for a card and reschedule all cards, they will only see the first revlog entry generated by `set due date`.

Reply:
This only affects card rescheduling when saving deck options, yes? Won't an upgrading user with previous set due date use see quite different results when rescheduling now, compared to what they were getting previously? Is that intentional?